plugins/module_utils/robot.py

```diff
@@ -82,28 +82,35 @@ def fetch_url_json(module, url, method='GET', timeout=10, data=None, headers=Non
     module.params['url_password'] = module.params['hetzner_password']
     module.params['force_basic_auth'] = True
     resp, info = fetch_url(module, url, method=method, timeout=timeout, data=data, headers=headers)
+    status_code = info["status"]
     try:
         content = resp.read()
     except AttributeError:
         content = info.pop('body', None)
 
-    if not content:
+    if not content and status_code != 200:
```

Comment:
```suggestion
    if not content:
```
Error handling is done below.

Reply:
A DELETE request returns status code 200 and no content, we will need some check here.

Reply:
I think it would be best to add a new parameter `allow_empty_result=False` (for which the delete call can pass `True`) which replaces content by `'{}'.encode('utf-8')` in this case instead of failing. Then the remainder of the function does not need to be modified.